serverless/cmd/sequence/main.go

```diff
@@ -51,22 +73,29 @@ func main() {
 	h := hasher.DefaultHasher
 	// init storage
 	var st *fs.Storage
-	if *create {
-		st, err = fs.Create(*storageDir, h.EmptyRoot())
-	} else {
-		var cpRaw []byte
-		cpRaw, err = fs.ReadCheckpoint(*storageDir)
-		if err != nil {
-			glog.Exitf("Failed to read log checkpoint: %q", err)
-		}
-		var cp *log.Checkpoint
-		if _, err := cp.Unmarshal(cpRaw); err != nil {
-			glog.Exitf("Failed to unmarshal checkpoint: %q", err)
-		}
-		st, err = fs.Load(*storageDir, cp)
+
+	cpRaw, err := fs.ReadCheckpoint(*storageDir)
+	if err != nil {
+		glog.Exitf("Failed to read log checkpoint: %q", err)
+	}
+
+	// Check signatures
+	v, err := note.NewVerifier(pubKey)
+	if err != nil {
+		glog.Exitf("Failed to instantiate Verifier: %q", err)
+	}
+	vCp, err := note.Open(cpRaw, note.VerifierList(v))
+	if err != nil {
+		glog.Exitf("Failed to open Checkpoint: %q", err)
+	}
+
+	var cp log.Checkpoint
+	if _, err := cp.Unmarshal([]byte(vCp.Text)); err != nil {
+		glog.Exitf("Failed to unmarshal checkpoint: %q", err)
 	}
+	st, err = fs.Load(*storageDir, &cp)
```

Comment:
With `create` gone, this can be `st, err :=` and remove the `var st *...` above

Reply:
done